tests/util_test.py

```diff
@@ -55,3 +57,103 @@ def test_SExpr_format():
 #     assert unescape_string('\\u3042') == 'あ'
 #     assert unescape_string('\\xe3\\x81\\x82') == 'あ'
 #     assert unescape_string('\\N{HIRAGANA LETTER A}') == 'あ'
+
+@pytest.fixture
+def empty_file(tmpdir):
+    f = tmpdir.join('empty.txt')
+    return str(f)
+
+
+@pytest.fixture
+def nocomment_file(tmpdir):
+    f = tmpdir.join('nocomment.txt')
+    f.write('avm := *top*.')
+    return str(f)
+
+
+@pytest.fixture
+def utf8_file(tmpdir):
+    f = tmpdir.join('utf8.txt')
+    f.write('; coding: utf-8\n'
+            u'a := character & [ ORTH \"あ\" ].')
+    return str(f)
+
+
+@pytest.fixture
+def utf8var1_file(tmpdir):
+    f = tmpdir.join('utf8var1.txt')
+    f.write('; -*- mode: tdl; encoding: UTF-8; foo: bar -*-\n'
+            u'a := character & [ ORTH \"あ\" ].')
+    return str(f)
+
+
+@pytest.fixture
+def utf8var2_file(tmpdir):
+    f = tmpdir.join('utf8var2.txt')
+    f.write('# coding: utf-8\n'
+            u'a=\"あ\"')
+    return str(f)
+
+
+@pytest.fixture
+def latin1_file(tmpdir):
+    f = tmpdir.join('latin1.txt')
+    f.write('; coding: iso-8859-1\n'
+            u'a := character & [ ORTH \"á\" ].')
+    return str(f)
+
+
+@pytest.fixture
+def shiftjis_file(tmpdir):
+    f = tmpdir.join('shift_jis.txt')
+    f.write('# coding: shift_jis\n'
+            u'a=\"あ\"')
+    return str(f)
+
+
+@pytest.fixture
+def eucjp_file(tmpdir):
+    f = tmpdir.join('eucjp.txt')
+    f.write('# coding: euc_jp\n'
+            u'a=\"あ\"')
+    return str(f)
+
+
+@pytest.fixture
+def invalid1_file(tmpdir):
+    f = tmpdir.join('invalid1.txt')
+    f.write('; encode: iso-8859-1\n'
+            'á')
+    return str(f)
+
+
+@pytest.fixture
+def invalid2_file(tmpdir):
+    f = tmpdir.join('invalid2.txt')
+    f.write('; coding: foo')
+    return str(f)
+
+import codecs
+@pytest.fixture
+def invalid3_file(tmpdir):
+    f = tmpdir.join('invalid3.txt')
+    f.write(codecs.BOM_UTF8 + '; coding: latin-1')
+    return str(f)
+
+@pytest.fixture
```

Comment:
The test function should not be made into a fixture. Currently the tests do not run because pytest does not see them as a function to test, just as a fixture.

Reply:
That explains why this stopped the errors I was running into. It seems that the pytest function opens the temp file with an ascii encoding, so I was getting errors when I tried to write non-ascii characters to it. Do you know how to resolve that?

Reply:
First, remove the `@pytest.fixture` so the tests run, then use `f.write_text(..., encoding='utf-8')` (also see my previous comment; this should be done for all temp files).

I also realized when looking at the docs that the `path.join(...)` isn't what opens the file, it's the `path.write()` itself, and `write()` does not accept an `encoding` parameter, but `write_text()` does.